lib/libspl/include/assert.h

```diff
@@ -40,6 +40,24 @@ libspl_assert(const char *buf, const char *file, const char *func, int line)
 	abort();
 }
 
+/* printf version of libspl_assert */
+static inline int
+libspl_assertf(const char *file, const char *func, int line, char *format, ...)
+{
+	char *buf = NULL;
+	int size;
+	va_list args;
+	va_start(args, format);
+	size = vsnprintf(NULL, 0, format, args);
+	buf = alloca(size + 1);
```

Comment:
What I meant was feed the format string directly to fprintf, so we don't need to allocate any thing.